src/um/winsock2.rs

```diff
@@ -165,30 +226,181 @@ STRUCT!{struct linger {
 }}
 pub const SOL_SOCKET: c_int = 0xffff;
 pub const SOMAXCONN: c_int = 0x7fffffff;
+#[inline]
+pub fn SOMAXCONN_HINT(b: c_int) -> c_int {
+    -b
+}
+pub const MSG_OOB: c_int = 0x1;
+pub const MSG_PEEK: c_int = 0x2;
+pub const MSG_DONTROUTE: c_int = 0x4;
+pub const MSG_WAITALL: c_int = 0x8;
+pub const MSG_PUSH_IMMEDIATE: c_int = 0x20;
+pub const MSG_PARTIAL: c_int = 0x8000;
+pub const MSG_INTERRUPT: c_int = 0x10;
+pub const MSG_MAXIOVLEN: c_int = 16;
+pub const MAXGETHOSTSTRUCT: usize = 1024;
+pub const FD_READ_BIT: c_long = 0;
+pub const FD_READ: c_long = 1 << FD_READ_BIT;
+pub const FD_WRITE_BIT: c_long = 1;
+pub const FD_WRITE: c_long = 1 << FD_WRITE_BIT;
+pub const FD_OOB_BIT: c_long = 2;
+pub const FD_OOB: c_long = 1 << FD_OOB_BIT;
+pub const FD_ACCEPT_BIT: c_long = 3;
+pub const FD_ACCEPT: c_long = 1 << FD_ACCEPT_BIT;
+pub const FD_CONNECT_BIT: c_long = 4;
+pub const FD_CONNECT: c_long = 1 << FD_CONNECT_BIT;
+pub const FD_CLOSE_BIT: c_long = 5;
+pub const FD_CLOSE: c_long = 1 << FD_CLOSE_BIT;
+pub const FD_QOS_BIT: c_long = 6;
+pub const FD_QOS: c_long = 1 << FD_QOS_BIT;
+pub const FD_GROUP_QOS_BIT: c_long = 7;
+pub const FD_GROUP_QOS: c_long = 1 << FD_GROUP_QOS_BIT;
+pub const FD_ROUTING_INTERFACE_CHANGE_BIT: c_long = 8;
+pub const FD_ROUTING_INTERFACE_CHANGE: c_long = 1 << FD_ROUTING_INTERFACE_CHANGE_BIT;
+pub const FD_ADDRESS_LIST_CHANGE_BIT: c_long = 9;
+pub const FD_ADDRESS_LIST_CHANGE: c_long = 1 << FD_ADDRESS_LIST_CHANGE_BIT;
+pub const FD_MAX_EVENTS: usize = 10;
+pub const FD_ALL_EVENTS: c_long = 1 << (FD_MAX_EVENTS as i32) - 1;
```

Comment:
Actually, is this cast even necessary? I see `Shl` implementations for *every* combination of two integer types.